### PR TITLE
Qpp pr2 test

### PR DESCRIPTION
**Pull request by AndrewFasano**

### CODE REVIEW

File: plugins/api.c [hunk 1]
```diff
@@ -52,6 +52,8 @@
 #endif
 #endif
 
+#define QPP_MINIMUM_VERSION 2 // TODO move to header
```
Comment (AndrewFasano) [inline plugins/api.c:55]: Could this go in plugin-qpp.h?

File: plugins/loader.c [hunk 2]
```diff
@@ -208,18 +208,39 @@ static int plugin_load(struct qemu_plugin_desc *desc, const qemu_info_t *info, E
...
             goto err_symbol;
-        } else if (version > QEMU_PLUGIN_VERSION) {
+        } else if (ctx->version > QEMU_PLUGIN_VERSION) {
             error_setg(errp, "Could not load plugin %s: plugin requires API version %d, but "
                        "this QEMU supports only up to version %d",
-                       desc->path, version, QEMU_PLUGIN_VERSION);
+                       desc->path, ctx->version, QEMU_PLUGIN_VERSION);
             goto err_symbol;
         }
+        else if (ctx->version < 2) {
```
Comment (AndrewFasano) [inline plugins/loader.c:223]: We should update the docs somewhere to describe plugin versions and V1 vs V2

File: plugins/loader.c [hunk 3]
```diff
@@ -265,6 +287,15 @@ static int plugin_load(struct qemu_plugin_desc *desc, const qemu_info_t *info, E
     return 1;
 }
 
+void plugin_add_qpp_cb(struct qemu_plugin_ctx *ctx, const char *name) {
+    struct qemu_plugin_qpp_cb *new_cb;
+    new_cb = qemu_memalign(qemu_dcache_linesize, sizeof(*new_cb));
+    memset(new_cb, 0, sizeof(*new_cb));
+    new_cb->name = name;
+    new_cb->plugin = ctx->name;
```
Comment (AndrewFasano) [inline plugins/loader.c:295]: Is the plugin name already stored in the qemu_plugin_ctx? Can we just drop this entirely?

File: plugins/plugin.h [hunk 4]
```diff
@@ -46,11 +46,23 @@ struct qemu_plugin_state {
     struct qht dyn_cb_arr_ht;
 };
 
+typedef void (*cb_func_t) (gpointer evdata, gpointer udata);
+
+struct qemu_plugin_qpp_cb {
+    const char *plugin;
+    const char *name;
+    cb_func_t registered_cb_funcs[32];
```
Comment (AndrewFasano) [inline plugins/plugin.h:54]: Should we use a typedef here, perhaps just using QEMU_PLUGIN_EV_MAX again? Or make a new one for QEMU_PLUGIN_QPP_MAX?

**Review by AndrewFasano — COMMENTED** [review]
Looks pretty good to me, couple of ideas for changes. Then we can work on splitting into a clean commit history.

File: plugins/loader.c (hunk 3 above)
Comment (lisi-w) [reply plugins/loader.c:295]: yeah I think this can be safely dropped

File: plugins/api.c (hunk 1 above)
Comment (lisi-w) [reply plugins/api.c:55]: I think moving this to plugins/plugin.h makes more sense

File: plugins/plugin.h (hunk 4 above)
Comment (lisi-w) [reply plugins/plugin.h:54]: I think reusing QEMU_PLUGIN_EV_MAX works

**Pull request closed without merging**